hooks/useClickOutside.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect } from 'react';
+import type { RefObject } from 'react';
+
+// eslint-disable-next-line no-unused-vars
+type TypeHandler = (_event: MouseEvent | TouchEvent) => void;
+
+type UseClickOutsideProps = {
+  ref: RefObject<HTMLElement>;
+  handler: TypeHandler;
+};
+
+export const useClickOutside = (props: UseClickOutsideProps) => {
```

Comment:
I will test that

Reply:
It's a good idea but it's create a lot of typing error. So we should keep it as it is

Reply:
It's creating typing errors because you probably don't understand how to create types properly.

I agree with @shanpriyan that creating the ref from the Hook is a good idea.

Reply:
Instead of just refuting ideas, could you share your error and let us work together to solve this? But I'm 99% sure the type errors are because of the ambiguity of the destination reference (lack of typing on `useRef` to match any HTMLElement).

This can either be done by providing a low-level element to `useRef<HTMLElement>` which can probably still give errors, or create a generic type:

```tsx
export const useClickOutise = <T extends HTMLElement>(props: UseClickOutsideProps) => {
  const elementRef = useRef<T>();


  return elementRef;
}

// on the Component
const ref = useClickOutside<HTMLDivElement>(() => {});

return <div ref={ref}>...</div>
```